src/server.c

```diff
@@ -3240,8 +3240,28 @@ static void propagateNow(int dbid, robj **argv, int argc, int target) {
     if (!shouldPropagate(target)) return;
 
     /* This needs to be unreachable since the dataset should be fixed during
-     * replica pause (otherwise data may be lost during a failover) */
-    serverAssert(!(isPausedActions(PAUSE_ACTION_REPLICA) && (!server.client_pause_in_transaction)));
+     * replica pause (otherwise data may be lost during a failover).
+     *
+     * Thought, there are exceptions:
+     *
+     * 1. We allow write commands that were queued up before and after to
+     *    execute, if a CLIENT PAUSE executed during a transaction, we will
+     *    track the state, the CLIENT PAUSE takes effect only after a transaction
+     *    has finished.
+     * 2. Primary loses a slot during the pause, deletes all keys and replicates
+     *    DEL to its replicas. In this case, we will track the state, the dirty
+     *    slots will be deleted in the end without affecting the data consistency.
```

Comment:
I think the replica will hold on to the orphaned data though. I think the replica will take over, not take over the slot mentioned (since someone else has it) and not trigger either of the paths that has extra data in it. 

You're right it's not a correctness issue though.

Reply:
Can you elaborate this a bit more here? I feel like i got the point, but i am afraid i will get it wrong, so before we merge it, i want to make sure i get it right.